tests/test_edgeql_delete.py

```diff
@@ -494,3 +498,50 @@ async def test_edgeql_update_in_conditional_bad_02(self):
                         (DELETE DeleteTest)
                     );
             ''')
+
+    async def test_edgeql_delete_abstract_01(self):
+        await self.con.execute(r"""
+            SET MODULE test;
+
+            INSERT DeleteTest { name := 'child of abstract 1' };
+            INSERT DeleteTest2 { name := 'child of abstract 2' };
+        """)
+
+        await self.assert_query_result(
+            r"""
+                WITH
+                    MODULE test,
+                    D := (
+                        DELETE
+                            AbstractDeleteTest
+                        FILTER
+                            .name ILIKE 'child of abstract%'
+                    )
+                SELECT D { name } ORDER BY .name;
+            """,
+            [{
+                'name': 'child of abstract 1'
+            }, {
+                'name': 'child of abstract 2'
+            }],
+        )
+
+    @test.xfail('DELETE trips up link target deletion constraint improperly')
```

Comment:
Do you want me to take a look at the follow-up, here?

Reply:
I've thought about it, and I'm not 100% sure it's a bug, because you can rewrite things as 

```
WITH ToDelete := (SELECT LinkingObject.link)
DELETE ToDelete
```

Perhaps we could special-case straight `DELETE LinkingObject.link`.

Reply:
Actually, wait, shouldn't both of these fail? The link isn't set to `ALLOW`.